### ISSUE TITLE
Tap to pay not working on iOS 18

### ISSUE THREAD

**Issue by manashathparia** [stripe/stripe-terminal-react-native#830]
**Describe the bug**
Tap to pay not working on iOS 18 devices consistently. iOS 17 and bellow devices seems to be working fine. 
However, if one connects to a terminal reader before connecting to tap to pay, it does seem to be working.

**To Reproduce**

Use tap to pay on iOS 18

**Expected behavior**
It should work consistently 

**Screenshots**

**Stripe Terminal React Native SDK version**

- ^0.0.1-beta.22

**Smartphone (please complete the following information):**

- Device: iPhone 13
- OS: iOS 18.0

**Additional context**

**Comment by manashathparia** [comment]
upon debugging using Xcode, it seems that the connectLocalMobileReader method is not resolving, causing the process to hang.

           const connectMobileReader = async (reader: any, onConnect?: any) => {
              console.log('Connecting to mobile reader:', reader);

            const { error } = await connectLocalMobileReader({
                  locationId: location.data?.data.stripe_id,
                  reader: reader || (mobileReader as Reader.Type),
            });

            if (error) {
                  log({
                        message: 'Error connecting to mobile reader',
                        error,
                  });
            }
      };

![Screenshot 2024-10-08 at 1 14 39 AM](https://github.com/user-attachments/assets/02c1fa5b-d757-4696-8d20-247655b697e9)
![Screenshot 2024-10-08 at 1 14 24 AM](https://github.com/user-attachments/assets/780f7dce-9185-4721-93c2-3aeb3fe61e4c)

**Comment by tim-lin-bbpos** [comment]
I've test on my end on an iOS 18 device using the dev-app(https://github.com/stripe/stripe-terminal-react-native/tree/main/dev-app) and unfortunately it works from connect to successful payment so I'm a bit confuse about the issue you mentioned.
It seems related to connect? and what is the error log you get? It seems no `Error connecting to mobile reader` according to your code.
And in your log it also contains BT connection related stuff, what is your steps?
Thank you in advance.

**Comment by tim-lin-bbpos** [comment]
Sorry it take a while to test which part the issue is and also discuss internally with different people, so far we still not able to reproduce by just changing connection method. 
Would it be possible for your end to test the same condition using our dev-app to see if it still happened? 
And the same time we'll also try to see if there's other possibility to this issue. Thanks.

**Comment by nazli-stripe** [comment]
Looking at our logs for the device in screenshots (serial `fa8572a6cf7ab50955f7c6837b8ef781574577fe8ebd09b2cd1471d14c62aee5`) the main error we see is
`Already_connected_to_a_reader._Disconnect_from_the_reader,_or_power_it_off_before_trying_again`. Just want to rule out you are aware of that and are handling it.

**Comment by manashathparia** [comment]
yes @nazli-stripe I am aware of it and it is handled in the function

**Comment by nazli-stripe** [comment]
hey @manashathparia can you follow these instructions and share the logs with us for when the connection issue happens? https://it-training.apple.com/tutorials/support/sup075/

**Comment by shivamshivk** [comment]
Hi @nazli-stripe @tim-lin-bbpos Is there any update with the logs we shared last week?

**Comment by henryx-stripe** [comment]
Hi, sorry for the delay, using the timestamp of the logs you shared, I see a log event of `reader_didFinishInstallingUpdate_error` when trying to connect, can you confirm if you are handling these errors?

I also do see a log line on our side that should not be emitted, I will continue to look into it.

**Comment by henryx-stripe** [comment]
Hi @shivamshivk following up on my last ask:
```
 I see a log event of reader_didFinishInstallingUpdate_error when trying to connect, can you confirm if you are handling these errors?
```

are you still seeing this issue?

**Comment by henryx-stripe** [comment]
Closing this issue due to inactivity, feel free to open a new one if the issue is still persisting
